### PR TITLE
feat: added user submenus

### PR DESCRIPTION
**Pull request by CzarekDryl**
Updated user menu component and added submenus to items with arrows.

image

### CODE REVIEW

File: src/components/common/Extensions/UserNavigation/partials/UserMenu/UserMenu.tsx [hunk 1]
```diff
@@ -29,102 +30,143 @@ const UserMenu: FC<UserMenuProps> = ({
...
-              {formatMessage({ id: 'connectWallet' })}
-            </Button>
-          </div>
-        </>
+      classNames={clsx(
+        'p-6 rounded-lg flex border-gray-100 justify-start z-[9999] tooltip-container relative overflow-hidden',
+        {
+          'w-full border-none shadow-none': isMobile,
+          'w-[20.125rem]': !isMobile,
+          'h-[32rem] md:h-[23rem]': !isWalletConnected && !activeSubmenu,
```
Comment (KatarzynaStefanowiczPagepro) [inline src/components/common/Extensions/UserNavigation/partials/UserMenu/UserMenu.tsx:46]: This style is used twice

File: src/components/common/Extensions/UserNavigation/partials/UserMenu/UserMenu.tsx [hunk 2]
```diff
@@ -11,6 +11,7 @@ import PopoverBase from '~shared/Extensions/PopoverBase';
 import TitledContent from '~common/Extensions/TitledContent/TitledContent';
 import WalletConnectedTopMenu from '../WalletConnectedTopMenu/WalletConnectedTopMenu';
 import Link from '~shared/Extensions/Link';
+import UserSubmenu from '../UserSubmenu/UserSubmenu';
```
Comment (KatarzynaStefanowiczPagepro) [inline src/components/common/Extensions/UserNavigation/partials/UserMenu/UserMenu.tsx:14]: import UserSubmenu from '../UserSubmenu';

And use shared/common import prefix

File: src/components/common/Extensions/UserNavigation/partials/UserMenu/UserMenu.tsx [hunk 3]
```diff
@@ -29,102 +30,143 @@ const UserMenu: FC<UserMenuProps> = ({
   const { formatMessage } = useIntl();
   const { connectWallet } = useAppContext();
   const { name, profile } = user || {};
+  const [activeSubmenu, setActiveSubmenu] = useState<string | null>(null);
```
Comment (KatarzynaStefanowiczPagepro) [inline src/components/common/Extensions/UserNavigation/partials/UserMenu/UserMenu.tsx:33]: isActiveSubmenu

File: src/components/common/Extensions/UserNavigation/partials/UserMenu/UserMenu.tsx [hunk 4]
```diff
@@ -29,102 +30,143 @@ const UserMenu: FC<UserMenuProps> = ({
...
+                <Link to="/" className="flex items-center justify-between">
+                  <div className="flex items-center">
+                    <Icon name="circles-three-plus" appearance={{ size: iconSize }} />
+                    <p className="ml-2">{formatMessage({ id: 'userMenu.getStartedTitle' })}</p>
+                  </div>
+                </Link>
+              </li>
+              <li className="mb-4">
+                <button
+                  type="button"
```
Comment (KatarzynaStefanowiczPagepro) [inline src/components/common/Extensions/UserNavigation/partials/UserMenu/UserMenu.tsx:118]: add aria-label

File: src/components/common/Extensions/UserNavigation/partials/UserSubmenu/UserSubmenu.tsx [hunk 5]
```diff
@@ -0,0 +1,24 @@
...
+import { userSubmenuItems } from './consts';
+import Icon from '~shared/Icon/Icon';
+
+const displayName = 'common.Extensions.UserNavigation.partials.UserSubmenu';
+
+const UserSubmenu: FC<UserSubmenuProps> = ({ submenuId }) => (
+  <ul className="mt-2">
+    {userSubmenuItems[submenuId].map((item) => (
+      <li key={item.id} className="mb-4 last:mb-0">
+        <Link to={item.url} className="flex items-center">
```
Comment (KatarzynaStefanowiczPagepro) [inline src/components/common/Extensions/UserNavigation/partials/UserSubmenu/UserSubmenu.tsx:13]: Shouldn't use src/components/shared/Extensions/NavLink/NavLink.tsx here?

File: src/components/common/Extensions/UserNavigation/partials/UserSubmenu/UserSubmenu.tsx [hunk 6]
```diff
@@ -0,0 +1,24 @@
+import React, { FC } from 'react';
+import { Link } from 'react-router-dom';
+import { UserSubmenuProps } from './types';
+import { userSubmenuItems } from './consts';
+import Icon from '~shared/Icon/Icon';
```
Comment (KatarzynaStefanowiczPagepro) [inline src/components/common/Extensions/UserNavigation/partials/UserSubmenu/UserSubmenu.tsx:5]: import Icon from '~shared/Icon';

File: src/components/common/Extensions/UserNavigation/partials/UserMenu/UserMenu.tsx (hunk 3 above)
Comment (CzarekDryl) [reply src/components/common/Extensions/UserNavigation/partials/UserMenu/UserMenu.tsx:33]: I'm setting subenu ID here and it's not boolean so I think isActiveSubmenu name doesn't fit

File: src/components/common/Extensions/UserNavigation/partials/UserMenu/UserMenu.tsx (hunk 1 above)
Comment (CzarekDryl) [reply src/components/common/Extensions/UserNavigation/partials/UserMenu/UserMenu.tsx:46]: One is for connected and second for not connected wallet -> `!isWalletConnected && !activeSubmenu` and `isWalletConnected && !activeSubmenu`

File: src/components/common/Extensions/UserNavigation/partials/UserSubmenu/UserSubmenu.tsx (hunk 6 above)
Comment (CzarekDryl) [reply src/components/common/Extensions/UserNavigation/partials/UserSubmenu/UserSubmenu.tsx]: Fixed

File: src/components/common/Extensions/UserNavigation/partials/UserSubmenu/UserSubmenu.tsx (hunk 5 above)
Comment (CzarekDryl) [reply src/components/common/Extensions/UserNavigation/partials/UserSubmenu/UserSubmenu.tsx]: Yeah, i forgot about that component

File: src/components/common/Extensions/UserNavigation/partials/UserMenu/UserMenu.tsx (hunk 4 above)
Comment (CzarekDryl) [reply src/components/common/Extensions/UserNavigation/partials/UserMenu/UserMenu.tsx:119]: aria-label added

File: src/components/common/Extensions/UserNavigation/partials/UserMenu/UserMenu.tsx (hunk 2 above)
Comment (CzarekDryl) [reply src/components/common/Extensions/UserNavigation/partials/UserMenu/UserMenu.tsx]: one `/UserSubmenu` removed but it's from partials folder so I think ../ isn't that bad

**Review by KatarzynaStefanowiczPagepro — APPROVED** [review]

File: src/components/common/Extensions/UserNavigation/partials/UserMenu/UserMenu.tsx [hunk 7]
```diff
@@ -29,102 +30,144 @@ const UserMenu: FC<UserMenuProps> = ({
...
+        })}
+      >
+        <button
+          type="button"
+          aria-label="Back to main menu"
+          className="flex items-center text-gray-400 text-xs font-medium"
+          onClick={() => setActiveSubmenu(null)}
+        >
+          <Icon name="caret-left" appearance={{ size: 'extraTiny' }} />
+          <p className="ml-2 uppercase">{activeSubmenu && formatMessage({ id: activeSubmenu })}</p>
```
Comment (krawczykkatarzyna) [inline src/components/common/Extensions/UserNavigation/partials/UserMenu/UserMenu.tsx:65]: activeSubmenu &&  `<p className="ml-2 uppercase">formatMessage({ id: activeSubmenu })}</p>`

File: src/components/common/Extensions/UserNavigation/partials/UserMenu/UserMenu.tsx [hunk 8]
```diff
@@ -29,102 +30,144 @@ const UserMenu: FC<UserMenuProps> = ({
...
+                  <div className="flex items-center">
+                    <Icon name="code" appearance={{ size: iconSize }} />
+                    <p className="ml-2">{formatMessage({ id: 'userMenu.developersTitle' })}</p>
+                  </div>
+                  <Icon name="caret-right" appearance={{ size: 'extraTiny' }} />
+                </button>
+              </li>
+              <li>
+                <button
+                  type="button"
```
Comment (krawczykkatarzyna) [inline src/components/common/Extensions/UserNavigation/partials/UserMenu/UserMenu.tsx:145]: aria-label missing

File: src/components/common/Extensions/UserNavigation/partials/UserMenu/UserMenu.tsx [hunk 9]
```diff
@@ -29,102 +30,144 @@ const UserMenu: FC<UserMenuProps> = ({
...
+                  <div className="flex items-center">
+                    <Icon name="lifebuoy" appearance={{ size: iconSize }} />
+                    <p className="ml-2">{formatMessage({ id: 'userMenu.contactAndSupportTitle' })}</p>
+                  </div>
+                  <Icon name="caret-right" appearance={{ size: 'extraTiny' }} />
+                </button>
+              </li>
+              <li className="mb-4">
+                <button
+                  type="button"
```
Comment (krawczykkatarzyna) [inline src/components/common/Extensions/UserNavigation/partials/UserMenu/UserMenu.tsx:132]: aria-label missing

File: src/components/common/Extensions/UserNavigation/partials/UserMenu/UserMenu.tsx [hunk 10]
```diff
@@ -29,102 +30,144 @@ const UserMenu: FC<UserMenuProps> = ({
...
+              <Button mode="quaternaryOutline" isFullSize onClick={connectWallet}>
+                {formatMessage({ id: 'connectWallet' })}
+              </Button>
+            </div>
+          </>
+        )}
+        <div className="w-full pb-6 mb-6 border-b border-b-gray-200 md:pb-5 md:mb-5">
+          <TitledContent title={{ id: 'userMenu.optionsTitle' }}>
+            <ul className="text-lg font-semibold md:font-normal md:text-md">
+              <li className="mb-4">
```
Comment (krawczykkatarzyna) [inline src/components/common/Extensions/UserNavigation/partials/UserMenu/UserMenu.tsx:109]: please create array with list

**Review by krawczykkatarzyna — APPROVED** [review]

**Pull request merged**